src/frontend/AppContainerWrapper.tsx

```diff
@@ -94,11 +98,43 @@ function hidePracticeModeTemporarily(route: string | null) {
 
 const AppContainerWrapper = () => {
   const navRef = React.useRef<NavigationContainerComponent>();
+  const appState = React.useRef(AppState.currentState);
   const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
   const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
   const [{ experiments }] = React.useContext(SettingsContext);
   const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
   const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
+  const {
+    killStateActive,
+    passcode,
+    checkFlag,
+    setCheckFlag,
+  } = React.useContext(SecurityContext);
+
+  React.useEffect(() => {
+    const StateListener = AppState.addEventListener(
```

Comment:
nit: camel case and rename to something more descriptive e.g. `appStateListener`